File: Assets/ToLua/Lua/user/ui/expedition/model/expedition_dungeon_info.lua
local t = {}
t.__index = t

local expedition_data = gamemanager.GetData('expedition_data')

function t.New(dataId,isFinished,isGetReward,isUnlocked)
  
  local o = {}
  setmetatable(o,t)
  o.id = dataId
  o.data = expedition_data.GetDataById(dataId)
  o.isFinished = isFinished
  o.isGetReward = isGetReward
  o.isUnlocked = isUnlocked
  
  return o
  
end

return t

File: Assets/Code/Logic/UI/Expedition/Model/ExpeditionDungeonInfo.cs
using UnityEngine;
using System.Collections;

namespace Logic.UI.Expedition.Model
{
	public class ExpeditionDungeonInfo 
	{
		public int id;
		public ExpeditionData data;
		public bool isFinished;
		public bool isGetReward;
		public bool isUnlocked;
		public ExpeditionDungeonInfo(int dataid,bool isFinished,bool isGetReward,bool isUnlocked)
		{
			Set(dataid,isFinished,isGetReward,isUnlocked);
		}
		public void Set(int dataid,bool isFinished,bool isGetReward,bool isUnlocked)
		{
			id = dataid;
			data = ExpeditionData.GetExpeditionDataByID(dataid);
			this.isFinished = isFinished;
			this.isGetReward = isGetReward;
			this.isUnlocked = isUnlocked;
		}
		public void SetFinished(bool isFinished)
		{
			this.isFinished = isFinished;
		}
		public void SetGetReward(bool isGetReward)
		{
			this.isGetReward = isGetReward;
		}
		public void SetUnlocked(bool isUnlocked)
		{
			this.isUnlocked = isUnlocked;
		}

	}
}

